### PR TITLE
Compile block arguments with defaults

### PR DESCRIPTION
**Pull request by elliottt**
Unify the handling of method and block arguments so that the existing machinery for detecting missing block arguments is the same.

### Motivation

Handling block arguments with defaults during code generation.

### Test plan

See included automated tests.

### CODE REVIEW

File: compiler/IREmitter/IREmitter.cc [hunk 1]
```diff
@@ -345,8 +346,8 @@ void setupArguments(CompilerState &base, cfg::CFG &cfg, const ast::MethodDef &md
                     }
 
                     // mark the arg as present
-                    auto &argPresent = irctx.argPresentVariables[i];
-                    if (!isBlock && argPresent.exists()) {
```
Comment (froydnj) [inline compiler/IREmitter/IREmitter.cc:349]: I just want to say how happy I am to see these weird `!isBlock` cases go away.

**Review by froydnj — APPROVED** [review]
This is amazingly straightforward.

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KCcIDXuSuira7a
→ https://go/builds/bui_KCcIvuuTefeMEc

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KCd3ftmpMPh3g1
→ https://go/builds/bui_KCd3Ra865t8fdp

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KCjMgoZXrMO1BE
→ https://go/builds/bui_KCjMZpSKhTzuFT

**Pull request merged**